### ISSUE TITLE
scrollbar covered cell content

### ISSUE THREAD

**Issue by anjianshi** [bvaughn/react-virtualized#222]
When columns width === grid width and there's a vertical scrollbar, the scrollbar will cover the last cell of every row.
Similarly, when rows Height === grid height and theres a horizontal scrollbar, it will cover the last cell of every column.

here's an example: http://plnkr.co/edit/lV9gaaS34xoybhfiWipJ

![a](https://cloud.githubusercontent.com/assets/5005012/15113063/acf553f6-1624-11e6-8571-de596f93ecfb.png)
![b](https://cloud.githubusercontent.com/assets/5005012/15113065/acfe2f9e-1624-11e6-9241-192f7080a37b.png)

Seems the problem was from Grid.js  [here](https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/Grid.js#L436) and [here](https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/Grid.js#L440)

May be change it to something like this can solve the problem?

```js
    if (totalColumnsWidth + (totalRowsHeight > height ? this._scrollbarSize : 0) <= width) {
      gridStyle.overflowX = 'hidden'
    }

    if (totalRowsHeight + (totalColumnsWidth > width ? this._scrollbarSize : 0) <= height) {
      gridStyle.overflowY = 'hidden'
    }
```

**Comment by bvaughn** [comment]
It's a little tricky to get right. There are edge-case sizes where everything would fit on one axis, then adding a scrollbar to the other causes things to overflow.

If you'd like to take a stab at it though feel free to submit a PR? :)

**Comment by anjianshi** [comment]
I think this solution has no problem.

About scrollbar, there's three situations:

1. `gridSize - contentSize >= scrollbarSize` (no scrollbar)
2. `contentSize > gridSize` (has scrollbar)
3. `gridSize > contentSize && gridSize - contentSize < scrollbarSize` (according to opposite axis)
    - opposite axis was in situation one (no scrollbar)
    - opposite axis was in situation two (has scrollbar)
   - opposite axis was in situation three too (both has no scrollbar)

So, if `gridSize >= contentSize + (oppositeAxisWillHaveScrollbarWhenThisAxisHasNoScrollbar ? scrollbarSize : 0)`, 
(same as `gridSize >= contentSize + (oppositeContentSize > oppositeGridSize ? scrollbarSize : 0)`) 
it will has no scrollbar, otherwish, it will has a scrollbar. 

I have submitted a PR for this. :)

**Comment by bvaughn** [comment]
Thanks for the PR @anjianshi ! I left some feedback.

**Comment by bvaughn** [comment]
This has been fixed in version 7.1.1. Thanks for your helpful suggestions and PR! (I gave you a shout-out in the release notes.) Cheers~
